src/net/socket_ancillary.rs

```diff
@@ -0,0 +1,777 @@
+// TODO: remove
+#![allow(dead_code, unused_variables)]
+#![allow(unsafe_code)]
+
+use core::convert::TryFrom;
+use core::marker::PhantomData;
+use core::mem::{size_of, zeroed};
+use core::ptr::{self, read_unaligned};
+use core::slice;
+
+use crate::imp::c;
+use crate::imp::fd::AsFd;
+use crate::imp::syscalls::{getgid, getpid, getuid};
+use crate::io::OwnedFd;
+use crate::process::{Gid, Pid, Uid};
+
+/// Create a buffer large enough for storing some control messages as returned by `recvmsg`.
+///
+/// # Examples
+///
+/// ```
+/// # fn main() {
+/// use rustix::{cmsg_space, net::SocketCred};
+/// use rustix::io::OwnedFd;
+/// // Create a buffer big enough for a `ScmRights` message with two file descriptors.
+/// let _ = cmsg_space!([OwnedFd; 2]);
+/// // Create a buffer big enough for a `ScmRights` message and a `ScmCredentials` message.
+/// let _ = cmsg_space!(OwnedFd, SocketCred);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! cmsg_space {
+    ( $( $x:ty ),* ) => {
+        {
```

Comment:
An idiom I've seen used in std for macros like this is to put these two `{`s together, like `=> {{`, with a `}}` at the end, so that the body only needs to be indented one level.

Reply:
good point, fixed